lib/y2logsstep.pm

```diff
@@ -66,22 +66,21 @@ sub get_to_console {
 # to workaround dependency issues
 sub workaround_dependency_issues {
     return unless check_screen 'dependency-issue', 10;
+    while (check_screen('dependency-issue', 5)) {
+        if (check_var('VIDEOMODE', 'text')) {
 
-    if (check_var('VIDEOMODE', 'text')) {
-        while (check_screen('dependency-issue', 5)) {
             wait_screen_change { send_key 'alt-s' };
             wait_screen_change { send_key 'ret' };
             wait_screen_change { send_key 'alt-o' };
         }
-    }
-    else {
-        while (check_screen('dependency-issue', 5)) {
+        else {
+            if (is_sle('15+') && check_var('PATTERNS', 'all') && check_screen('missing-dependency', 0)) {
+                record_soft_failure 'bsc#1093073 - Missing dependencies for KDE patterns in selection patterns';
```

Comment:
How do you know at this point that it's this bug? I would rather suggest to just add the "workaround" property to the needle and not add this record_soft_failure here. An alternative could be to add a specific tag to the needle for which you check, e.g. just the tag 'bsc#1093073'